employee-rostering-frontend/cypress/integration/newTenantBasicWorkflow.js

```diff
@@ -0,0 +1,166 @@
+        cy.get('[data-cy=publish-notice]').type("7");
+        // publish length is disabled/not supported by backend
+        // cy.get('[data-cy=publish-length]').type("7");
+        cy.get('[data-cy=rotation-length]').type("7");
+        selectValue("Select a timezone...", "UTC");
+        cy.get('[data-cy=save]').click();
+        closeAlerts();
+        
+        changeToTenant("Test Tenant");
+        
+        // Create a new skill
+        gotoPage("skills");
+        cy.get("button").contains("Add").click();
+        cy.get('[aria-label="Name"]').type("New Skill");
+        cy.get('[aria-label="Save"]').click();
+        closeAlerts();
+        
+        // Create two spots
+        gotoPage("spots");
+        
+        // First spot doesn't require any skills
+        cy.get("button").contains("Add").click();
+        cy.get('[aria-label="Name"]').type("No Skill Spot");
+        cy.get('[aria-label="Save"]').click();
+        closeAlerts();
+        
+        // Second spot requires our skill
+        cy.get("button").contains("Add").click();
+        cy.get('[aria-label="Name"]').type("Required Skill Spot");
+        selectValue("Select required skills...", "New Skill");
+        cy.get('[aria-label="Save"]').click();
+        closeAlerts();
+        
+        // Create a contract
+        gotoPage("contracts");
+        cy.get("button").contains("Add").click();
+        cy.get('[aria-label="Name"]').type("New Contract");
+        cy.get('[aria-label="Save"]').click();
+        closeAlerts();
+        
+        // Create two employees
+        gotoPage("employees");
+        
+        // First employee doesn't have any skills
+        cy.get("button").contains("Add").click();
+        cy.get('[aria-label="Name"]').type("No Skills Employee");
+        selectValue("Select a contract...", "New Contract");
+        cy.get('[aria-label="Save"]').click();
+        
+        // Second employee has our skill
+        cy.get("button").contains("Add").click();
+        cy.get('[aria-label="Name"]').type("Employee with Skills");
+        selectValue("Select a contract...", "New Contract");
+        selectValue("Select skill proficiencies...", "New Skill");
+        cy.get('[aria-label="Save"]').click();
+        closeAlerts();
+        
+        // Create some shifts
+        gotoPage("shift");
+        
+        // First shift is on Monday, and is for the spot that doesn't require any skills
+        selectValue("Select a Spot...", "No Skill Spot");
+        cy.wait(1000);
+        dragCreateShift(1, "09:00", "17:00");
+        
+        // Second shift is in the early morning on Tuesday, and is for the spot that requires our skill
+        selectValue("Select a Spot...", "Required Skill Spot");
+        cy.wait(1000);
+        dragCreateShift(2, "00:00", "06:00");
+        
+        // Schedule for 5 seconds
+        cy.get('[aria-label="Actions"]').click();
+        cy.contains("Schedule").click();
+        cy.wait(5000);
```

Comment:
This is for OptaPlanner to provide a solution, right? Is there any request we could potentially wait for instead of a hard-coded wait for 5 seconds?

Reply:
In theory, the first or second GET of the current shift roster should have the updated shifts, but I think a wait is not too bad here (the app waits for 1-5 seconds before sending the next GET anyways).

Reply:
I am fine with the fact that it might take 5 seconds - here is makes sense. The only thing which worries me is what happens on a slower machine - if there is no solution sent from server within 5 seconds, it's going to fail.

Reply:
I don't think any QE enviroments use Babbage Machines nowadays, so even 5 seconds is overkill (We have 2 shifts and 2 employees, or 9 possibilities; for the server to not come up with a solution in 5 seconds mean we have a score calculation speed less than 2/s) .

Reply:
Truth, Babbage machines are difficult to provision.

Let's see if the test is stable over time and improve this part if needed.